### PR TITLE
Adding ApiConsistencyTest for *.Design projects

### PR DESCRIPTION
**Pull request by lajones**
Depends on PR#1646

Adding ApiConsistencyTest for both EntityFramework.Relational.Design and EntityFramework.SqlServer.Design and updated the code to match (virtual's, [NotNull]'s).

Plus removed a reference to the no-longer-existent EntityFramework.ReverseEngineering project in EntityFramework-2014.sln.

### CODE REVIEW

File: src/EntityFramework.Relational.Design/CodeGeneration/CSharpCodeGeneratorHelper.cs [hunk 1]
```diff
@@ -5,26 +5,27 @@
...
 using System.Reflection;
+using JetBrains.Annotations;
 using Microsoft.Data.Entity.Utilities;
 
 namespace Microsoft.Data.Entity.Relational.Design.CodeGeneration
 {
     public class CSharpCodeGeneratorHelper
     {
-        public virtual void SingleLineComment(string comment, IndentedStringBuilder sb)
+        public virtual void SingleLineComment([NotNull] string comment, [NotNull] IndentedStringBuilder sb)
```
Comment (bricelam) [inline src/EntityFramework.Relational.Design/CodeGeneration/CSharpCodeGeneratorHelper.cs]: Should also add `Check.NotNull()` calls.

**Comment by bricelam** [comment]
:shipit:

File: src/EntityFramework.Relational.Design/CodeGeneration/CSharpCodeGeneratorHelper.cs (hunk 1 above)
Comment (lajones) [inline src/EntityFramework.Relational.Design/CodeGeneration/CSharpCodeGeneratorHelper.cs]: Will do. Thanks.

**Comment by lajones** [comment]
Updated and checked in - commit f4e5717.

**Pull request closed without merging**